### PR TITLE
Maya: Collect Review cleanup code for attached subsets

### LINKED ISSUE #4718: Maya: Collect review refactor
The logic for `reviewable_subset` definitely feels off. Instead of `for inst in instance.context` with incrementing i += 1 and then accessing that instance again through `instance.context[i]` feels completely invalid. As far as I know `instance.name` isn't enforced to be unique at all in pyblish so the logic here would even break if it finds more matches since the `i` index would be off.

It feels more like this should've been:

```python
reviewable_inst = next(inst for inst in context if inst.name == reviewable_subset)
data = reviewable_inst.data
```

Also `instance.context[i].data.update(data)` is completely redundant since it's changing that data dictionary already in-place.
Not relevant to this PR I'd say but we should clean that up soon!

_Originally posted by @BigRoy in https://github.com/ynput/OpenPype/pull/4716#pullrequestreview-1358970933_
            

[cuID:[OP-5385](https://app.clickup.com/t/861mjun0b)]

### PR DESCRIPTION
**Pull request by BigRoy**
## Changelog Description

Refactor some code for Maya: Collect Review for attached subsets.

## Additional info

Resolves #4718

## Testing notes:
1. Publish a review that attaches to another subset (add e.g. `modelMain` instance into `reviewMain`)
2. To be sure also publish a review that doesn't attach to another subset.

### CODE REVIEW

**Review by LiborBatek — DISMISSED** [review]
Tested both scenarios and both works normally and correctly output review within the used instance and or outside as a separate review publish.

image

All works as expected.

**Review by tokejepsen — APPROVED** [review]
Tested in Maya 2023 and 2020 successfully. Code looks good as well.

File: openpype/hosts/maya/plugins/publish/collect_review.py [hunk 1]
```diff
@@ -35,57 +34,68 @@ def process(self, instance):
...
+            data['step'] = instance.data['step']
+            data['fps'] = instance.data['fps']
+            data['review_width'] = instance.data['review_width']
+            data['review_height'] = instance.data['review_height']
+            data["isolate"] = instance.data["isolate"]
+            data["panZoom"] = instance.data.get("panZoom", False)
+            data["panel"] = instance.data["panel"]
+
+            # The review instance must be active
+            cmds.setAttr(str(instance) + '.active', 1)
```
[the remark this answers is not in the document]
Comment (jakubjezek001) [reply openpype/hosts/maya/plugins/publish/collect_review.py:96]: Shell this be resolved with different PR, lets do an issue from this so we can merge this, please.

[the remark this answers is not in the document]
Comment (BigRoy) [reply openpype/hosts/maya/plugins/publish/collect_review.py:96]: Yes, let's do it with a separate issue. I'm not sure what that logic intended to do.

**Review by LiborBatek — APPROVED** [review]
Again tested and works fine.
image

Also works ok when no nested instances present inside of ```Review``` Instance...

**Pull request merged**